getStartedProgGuide.md

```diff
@@ -0,0 +1,260 @@
+[.Net](../getStarted.md) | [PHP](../getStarted.md) | [Java](../getStarted.md) | [Python](../getStarted.md) | [Node-JS](../getStarted.md) | [Ruby](../getStarted.md) | [Go](../getStarted.md)
+
+
+### Core Concepts
+#### REST Calls and Callbacks
+Bandwidth's Voice & Messaging APIs use callbacks to communicate with the developer's servers hosting applications.  Callbacks are triggered by things like incoming calls, incoming messages, or other API calls that have a callback defined.  Understadning callbacks and how they work is a prerequisite to developing complex communications applications with Bandwidth's APIs.
+
+![Callback Flow](https://dev.bandwidth.com/ap-docs/apiCallbacks/images/callback-events1.png)
+
+*Guides*
+* [Send SMS & MMS Guide](https://dev.bandwidth.com/howto/sendSMSMMS.html)
+* [Receive Messages or Calls](https://dev.bandwidth.com/howto/incomingCallandMessaging.html) 
+* [Building Blocks](https://dev.bandwidth.com/howto/howto.html) 
+* [Rate Limits](https://dev.bandwidth.com/ap-docs/rateLimits.html) 
+
+#### BXML for Voice Applications
+Voice applications can be built using BXML (Bandwidth Extensible Markup Language).  BXML is incredibly powerful and easy to use.  For incoming calls, your application can form and serve BXML documents on the URL setup for voice callbacks.  For outgoing voice calls, a REST API is used to establish the call, then the callback will fetch the BXML from your application server.
+
+![BXML Flow](https://dev.bandwidth.com/ap-docs/bxml/BXMLCallFlow.png)
+
+*Guides*
+* [BXML Overview](https://dev.bandwidth.com/ap-docs/bxml/bxmlOverview.html)
+* [BXML Guide](https://dev.bandwidth.com/ap-docs/bxml/bxmlConcepts.html)
+* [BXML Reference](https://dev.bandwidth.com/ap-docs/bxml/bxml.html)
+
+### Step-by-Step Voice & Messaging Development
+#### Development Environment and Authentication Setup 
+
+*_The examples below use C# and Bandwidth's [.Net SDK](https://dev.bandwidth.com/clientLib/csharp.html), but the basic steps are constistent in most any programming language.  The links above point to examples specific to those programming languages and the menu to the left conatins links to SDKs in various programming languages.  You are not required to use an SDK or specific HTTP client, you can consume Bandwidth's APIs directly using any programming language or tool capable of making HTTPS requests._*
+
+* Open your favorite .Net/C# IDE.  You can find a free one [here](https://visualstudio.microsoft.com/vs/community/).   
+* Get the Bandwidth C# SDK (Available via NuGet or [here](https://dev.bandwidth.com/clientLib/csharp.html)) 
+* Include necessary libraries.
+
+```csharp
+using System;
+using System.Threading.Tasks;
+using Bandwidth.Net;
+using Bandwidth.Net.Api;
+```
+
+* Create a class and setup authentication constants.  NOTE: In a production setting, the authentication strings should be stored in a secure location such as environment variables.  
+
+```csharp
+public class Program
```

Comment:
Would recommend anywhere that we put user-id, token, secret that instead of strings we put the values as environment variables.

That way we don't have any places that encourage bad security practice.  If someone wants to delete the `Process.env` code and add a string with their token/secret we at least didn't provide that as source.